### PR TITLE
added FIFO cache metrics for current number of entries and memory usage

### LINKED ISSUE #2258: FIFO cache does not report number of entries and memory in use
FIFO cache has multiple useful metrics, but it lacks metrics for the current number of entries and used memory.

### PR DESCRIPTION
**Pull request by dmitsh**
fixed bug in updating last element of FIFO
updated unit tests

Signed-off-by: Dmitry Shmulevich <dmitry.shmulevich@sysdig.com>

**What this PR does**:
Adding metrics for current number of entries and memory usage in FIFO cache
Fixing corner case bug in FIFO (updating last element in FIFO)

**Which issue(s) this PR fixes**:
Fixes #2258 

**Checklist**
- [x ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/chunk/cache/fifo_cache_test.go [hunk 1]
```diff
@@ -72,18 +75,55 @@ func TestFifoCache(t *testing.T) {
...
 
-func TestFifoCacheExpiry(t *testing.T) {
-	c := NewFifoCache("test", FifoCacheConfig{Size: size, Validity: 5 * time.Millisecond})
+func TestFifoCacheEvictionExpiry(t *testing.T) {
+	c := NewFifoCache("test2", FifoCacheConfig{Size: 3, Validity: 5 * time.Millisecond})
 	ctx := context.Background()
+	memorySz := int(unsafe.Sizeof(cacheEntry{})) * 3
 
-	c.Put(ctx, []string{"0"}, []interface{}{0})
+	assert.Equal(t, testutil.ToFloat64(c.entriesAdded), float64(0))
```
Comment (gouthamve) [inline pkg/chunk/cache/fifo_cache_test.go]: Can we use `require.Equal` here not to add one more dependency? Or use this asset package: https://godoc.org/github.com/stretchr/testify/assert

Comment (pracucci) [reply pkg/chunk/cache/fifo_cache_test.go]: `assert` is more appropriate, but as Goutham outlined the wrong package was imported. The correct one is:
`github.com/stretchr/testify/assert`

Please cleanup `vendor/` and `go.mod` once done.

File: pkg/chunk/cache/fifo_cache.go [hunk 2]
```diff
@@ -213,6 +244,9 @@ func (c *FifoCache) put(ctx context.Context, key string, value interface{}) {
 	c.entries[c.last].next = index
 	c.first = index
 	c.index[key] = index
+
+	c.memoryBytes.Add(float64(sizeOf(key) + sizeOf(value)))
+	c.entriesCurrent.Set(float64(len(c.entries)))
```
Comment (pracucci) [inline pkg/chunk/cache/fifo_cache.go]: [nit, no strong opinions] Is it equal to do `c.entriesCurrent.Inc()` or am I wrong? If I'm correct, `.Inc()` may be just clearer.

File: pkg/chunk/cache/fifo_cache.go [hunk 3]
```diff
@@ -240,3 +274,32 @@ func (c *FifoCache) Get(ctx context.Context, key string) (interface{}, bool) {
...
+	case []int32:
+		return len(v) * 4
+	case []uint32:
+		return len(v) * 4
+	case []float32:
+		return len(v) * 4
+	case []int:
+		return len(v) * 8
+	case []uint:
+		return len(v) * 8
```
Comment (pracucci) [inline pkg/chunk/cache/fifo_cache.go]: FYI, these are machine dependent. It's not safe to assume it's `8` bytes, even if `8` is the most common case (ie. we don't release 32bit binaries and probably any production system runs on 64 bit). That being said, if we keep it like this I would add a comment to explain it.

File: pkg/chunk/cache/fifo_cache.go [hunk 4]
```diff
@@ -289,16 +289,23 @@ func sizeOf(i interface{}) int {
...
-		return len(v) * 8
-	case []uint:
-		return len(v) * 8
 	case []int64:
 		return len(v) * 8
 	case []uint64:
 		return len(v) * 8
 	case []float64:
 		return len(v) * 8
+	// next 2 cases are machine dependen
```
Comment (pracucci) [inline pkg/chunk/cache/fifo_cache.go]: `dependen` > `dependent`

File: pkg/chunk/cache/fifo_cache.go [hunk 5]
```diff
@@ -33,7 +33,7 @@ var (
 		Help:      "The total number of evicted entries",
 	}, []string{"cache"})
 
-	cacheEntriesCurrent = promauto.NewGaugeVec(prometheus.GaugeOpts{
+	cacheEntriesCurrent = promauto.NewCounterVec(prometheus.CounterOpts{
```
Comment (pracucci) [inline pkg/chunk/cache/fifo_cache.go]: The fact that currently the number of items in the cache can't decrease is just an implementation detail. Being an LRU cache and having the concept of eviction, in the future we may actually have a routine deleting expired items. For this reason, I would suggest to revert this back to a gague.

File: pkg/chunk/cache/fifo_cache.go (hunk 4 above)
Comment (dmitsh) [reply pkg/chunk/cache/fifo_cache.go]: thanks :)

File: pkg/chunk/cache/fifo_cache.go (hunk 5 above)
Comment (dmitsh) [reply pkg/chunk/cache/fifo_cache.go]: agree!
